### PR TITLE
Add documentation for IPC API

### PR DESCRIPTION
**Pull request by vinser52**
### Description

Create documentation for IPC API.
Fixes referencing between documents.

Ref. #404

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly
- [x] Extended the README/documentation

### CODE REVIEW

File: scripts/docs_config/introduction.rst [hunk 1]
```diff
@@ -134,6 +134,26 @@ An example of an environment variable for setting up logger with logging level s
...
+When a client requests an IPC handle for a memory allocated by UMF, UMF does the following:
+
+#. finds the corresponding memory pool.
+
+#. finds corresponding coarse grain allocation (multiple fine grain allocations might reside in a coarse grain memory region returned by the memory provider).
+
+#. checks if the IPC handle for a coarse grain is already cached and returns the cached handle if it exists. Otherwise creates a new IPC handle for the coarse grain allocation.
+
+Not every memory provider can and must support IPC operations. It is up to the memory provider implementation to decide if it supports IPC operations. 
+If the corresponding memory provider does not support IPC operations, UMF will return an error when a client requests an IPC handle for a memory object allocated by this memory provider.
```
Comment (lukaszstolarczuk) [inline scripts/docs_config/introduction.rst:156]: should we mention somewhere (perhaps in provider's description) which one supports this API?

**Review by lukaszstolarczuk — COMMENTED** [review]
do you have a preview of generated version of this docs?

// you'd have to enable GH Pages building and publishing on your fork and push this commit into your main

**Comment by vinser52** [comment]
> do you have a preview of generated version of this docs?

I build them locally on my machine.

> you'd have to enable GH Pages building and publishing on your fork and push this commit into your main.

Do you have exact instructions on what should be done?

**Comment by lukaszstolarczuk** [comment]
> > do you have a preview of generated version of this docs?
> 
> I build them locally on my machine.

I guess it should be ok, then

> > you'd have to enable GH Pages building and publishing on your fork and push this commit into your main.
> 
> Do you have exact instructions on what should be done?

Anyway, if you want to build them on your fork, all you have to do is:
- go to your fork's settings
- section `Pages`
- set source to `GitHub Actions` (not "deploy from a branch")
- push changes into the main branch, which should trigger GH Pages workflow (for me it's [here](https://github.com/lukaszstolarczuk/unified-memory-framework/actions/workflows/docs.yml) )

File: scripts/docs_config/introduction.rst (hunk 1 above)
Comment (vinser52) [reply scripts/docs_config/introduction.rst:156]: After thinking more about it, I have the following proposal:

- Today  [Memory provider API](https://oneapi-src.github.io/unified-memory-framework/api.html#providers) section contains only APIs from the `memory_provider.h` header.
- But also we should describe the `umf_memory_provider_ops_t` structure. It contains mandatory and optional interfaces (callbacks to implementation).
- Each memory provider implementation should describe which interfaces it implmenets. For example, L0 provider implements IPC but does not implement `umf_memory_provider_ext_ops_t`.

If you agree I will create an issue for that and it should be done in a separate PR.

Comment (lukaszstolarczuk) [reply scripts/docs_config/introduction.rst:156]: yeah, of course, that sounds great and we should do it in a sep. PR for all providers

Comment (vinser52) [reply scripts/docs_config/introduction.rst:156]: Done. ref. #434

**Review by lukaszstolarczuk — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**